### DISCUSSION TITLE
How to Automatically Add Rows When Pasting Data in ReactGrid?

### DISCUSSION THREAD

**Discussion by deadblue22** [silevis/reactgrid#190] [General]
I'm currently working on a project that involves using ReactGrid, and I'm seeking assistance with a specific functionality. I need help implementing a feature that automatically adds rows when pasting data into the grid.

I have gone through the ReactGrid documentation and searched online, but I haven't found a clear solution for this issue. I would greatly appreciate any insights, code snippets, examples, or suggestions that could help me achieve this.

The goal is to enhance the user experience by allowing them to paste a large amount of data into the grid and have it dynamically create rows to accommodate the pasted content. This would eliminate the need for manual row addition after each paste operation.

**Comment by djpirra** [comment]
Would be very keen to understand how this is achieved as well.

Thank you

**Comment by Michota** [comment]
If I did understand your question correctly, you want to extend the grid, if data that is pasted overextends grid? Well, if I am right, then [this feature won't be built-in ReactGrid](https://github.com/silevis/reactgrid/issues/280), though you may add it by yourself, in outer function.

**Reply by Anderbone** [reply]
Is there any plan to add it to v5 soon? Very keen to have this feature. Thank you!

**Comment by Anderbone** [comment]
I managed to do a quick demo 
https://github.com/silevis/reactgrid/discussions/474

**Comment by qiufeihong2018** [comment]
@deadblue22 I created a PR and completed this function (extend paste range), https://github.com/datafe/reactgrid/pull/3/files
